apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -181,6 +207,32 @@ export async function calculatePoolStats({
     ),
   ]);
 
+  if (results[0].status === "fulfilled" && !results[0].value?.pool) {
+    throw new Error(
+      `No pool with ID ${poolId}(${network}) in range ${startAtTimestamp} - ${endAtTimestamp}`,
+    );
+  }
+
+  if (results.some((p) => p.status === "rejected")) {
+    const errors = results
+      .filter((p) => p.status === "rejected")
+      // @ts-ignore
+      .map((p) => p.reason);
+    throw new Error(
+      `Error fetching data for pool ${poolId}(${network}) in range ${startAtTimestamp} - ${endAtTimestamp}: ${errors}, function ${errors[0]?.stack}}`,
+    );
+  }
+
+  const [
+    _,
+    balPriceUSD,
+    [tvl, volume, symbol, tokenBalance],
+    votingShare,
+    [feeAPR, collectedFeesUSD],
+    tokensAPR,
+    // @ts-ignore
+  ] = results.filter((p) => p.status === "fulfilled").map((p) => p.value);
```

Comment:
this is an (incomplete) attempt to make our querying logic more robust by allowing promise rejections not to throw the whole thing off. Ideally we'd gracefully handle them here and assign some default values and refetch the next time this page gets triggered.